### ISSUE TITLE
Spark SQL Extensions: MIGRATE command

### ISSUE THREAD

**Issue by aokolnychyi** [apache/iceberg#1591]
One should be able to use the MIGRATE command to migrate existing tables to Iceberg. Similar to SNAPSHOT, it should use the existing table definition to create a new Iceberg table and generate metadata for existing files. Apart from that, it should either swap the table pointer in the original catalog or rename the original table to a backup table (depending on circumstances, let's discuss). Once the table has been migrated to Iceberg, all writes and reads have to be done through Iceberg. In other words, the original table should no longer be accessible to non-Iceberg readers.

```
MIGRATE TABLE t [AS t2]
USING iceberg
[TBLPROPERTIES ('key' 'value')]
```

In query engines like Spark where we have a notion of a custom catalog, we may not always be able to swap a pointer in the original catalog as the source and target catalogs may be different. For example, if you want to move a regular Spark table that stores a pointer in the HMS to the Iceberg Hadoop catalog. For such cases we may want to consider exposing AS target.

Important to note that MIGRATE should inherit the location of the original table. New files must be written in the same layout. For example, Iceberg must set the data location as the root table location whenever migrating an existing dataset as opposed to having a separate data folder. Users should be prohibited from modifying the data location.

**Comment by aokolnychyi** [comment]
@RussellSpitzer is looking into that.

**Comment by waterlx** [comment]
Is this issue addressed by PR #1525? @aokolnychyi @RussellSpitzer

**Comment by RussellSpitzer** [comment]
I will be making a subsequent PR that acknowledge adds the SQL but that PR is going to be the programmatic api which the command will use

**Comment by liukun4515** [comment]
> Is this issue addressed by PR #1525? @aokolnychyi @RussellSpitzer

I think  #1525 just modify the action to support spark3( multi catalog).

**Comment by aokolnychyi** [comment]
This was done in PR #1906.
